axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/AccountingCutOffServiceImpl.java

```diff
@@ -323,9 +350,20 @@ public Move generateCutOffMove(
 
     this.generatePartnerMoveLine(move, origin, partnerAccount, moveDescription, originDate);
 
+    // Status
     if (move.getMoveLineList() != null && !move.getMoveLineList().isEmpty()) {
       move.setStockMove(stockMove);
-      moveValidateService.accounting(move);
+
+      if (cutOffMoveStatusSelect == MoveRepository.STATUS_SIMULATED) {
+        moveSimulateService.simulate(move);
+      } else {
+        moveValidateService.updateValidateStatus(
+            move, cutOffMoveStatusSelect == MoveRepository.STATUS_DAYBOOK);
+
+        if (cutOffMoveStatusSelect == MoveRepository.STATUS_ACCOUNTED) {
+          moveValidateService.accounting(move);
+        }
+      }
```

Comment:
Proposition :
* as for generateCutOffMoveFromMove, use CollectionUtils to have a null safe check if empty
* add a new method updateStatus, that way you can use it also in method `generateCutOffMoveFromMove`
```
if (CollectionUtils.isNotEmpty(cutOffMove.getMoveLineList())) {
   move.setStockMove(stockMove);
   updateStatus(move, cutOffMoveStatusSelect)
}
...
protected void updateStatus(Move move, int cutOffMoveStatusSelect) {

if (cutOffMoveStatusSelect == MoveRepository.STATUS_SIMULATED) {
        moveSimulateService.simulate(move);
      } else {
        moveValidateService.updateValidateStatus(
            move, cutOffMoveStatusSelect == MoveRepository.STATUS_DAYBOOK);

        if (cutOffMoveStatusSelect == MoveRepository.STATUS_ACCOUNTED) {
          moveValidateService.accounting(move);
        }
}
````